src/BlockchainManager.ts

```diff
@@ -0,0 +1,325 @@
+import { Resolver } from "did-resolver";
+import Web3 from "web3";
+
+const { Credentials } = require("uport-credentials");
+const { createVerifiableCredential, verifyCredential } = require("did-jwt-vc");
+const DidRegistryContract = require("ethr-did-registry");
+const didJWT = require("did-jwt");
+const { delegateTypes, getResolver } = require("ethr-did-resolver");
+const EthrDID = require("ethr-did");
+
+const blockChainSelector = (
+  networkConfig: { name: string; rpcUrl: string; registry: string }[],
+  did: string
+) => {
+  let routerCharPos = -1,
+    index = -1,
+    i = 1;
+  let searchArray = true;
+  const noUportPrefixDid = did.slice(9, did.length);
+  routerCharPos = noUportPrefixDid.search(":"); // if routerCharPos > 0 there's another prefix, should search the provider array
+  if (routerCharPos === -1) {
+    // if not, connect directly to mainnet
+    searchArray = false;
+    index = 0;
+  }
+  while (i < networkConfig.length && searchArray) {
+    routerCharPos = did.search(networkConfig[i].name);
+    if (routerCharPos > 0) {
+      index = i; // saves index for connection later
+      searchArray = false;
+    } else i++; // provider not found, keep going through array
+  }
+
+  let blockchainToConnect = {
+    provider: null,
+    address: null,
+  };
+
+  if (index >= 0) {
+    blockchainToConnect.provider = networkConfig[index].rpcUrl;
+    blockchainToConnect.address = networkConfig[index].registry;
+    return blockchainToConnect;
+  } else {
+    throw "Invalid Provider Prefix";
+  }
+};
+
+const addPrefix = function (prefixToAdd, did) {
+  const prefixedDid = did.slice(0, 9) + prefixToAdd + did.slice(9, did.length);
+  return prefixedDid;
+};
+
+const checkPrefix = function (prefix, networkArray) {
+  let i = 0, notFounded = true;  
+  while (i < networkArray.length && notFounded) {  
+    if (prefix === networkArray[i].name)
+      notFounded = false;
+    else {
+      i++;
+    }
+  }
+  return !notFounded;
+};
+
+interface BlockchainManagerConfig {
+  gasPrice: number;
```

Comment:
gasPrice se utilza?

Reply:
veo q se utiliza aca:

```
    options.gasPrice = await this.getGasPrice(web3);
    const delegateMethodSent = await addDelegateMethod.send(options);    
```

Reply:
La interfaz `BlockchainManagerConfig` tiene un `gasPrice` que le pasa al constructor de la clase, que luego el valor se almacena internamente como `this.config.gasPrice`. 
`options.gasPrice` es tipo `Options` que tiene definido también `gasPrice`. 
A pesar de tener el mismo nombre, no son lo mismo